packages/xdg_directories/lib/xdg_directories.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library xdg_directories;
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/cupertino.dart';
+import 'package:path/path.dart' as path;
+import 'package:process/process.dart';
+
+/// An override function used by the tests to override the environment variable
+/// lookups using [xdgEnvironmentOverride].
+typedef EnvironmentOverride = String Function(String envVar);
+
+/// A testing setter that replaces the real environment lookups with an override.
+///
+/// Set to null to stop overriding.
+///
+/// Only available to tests.
+@visibleForTesting
+set xdgEnvironmentOverride(EnvironmentOverride override) {
+  _xdgEnvironmentOverride = override;
+  _getenv = _xdgEnvironmentOverride ?? _productionGetEnv;
+}
+
+/// A testing getter that returns the current value of the override that
+/// replaces the real environment lookups with an override.
+///
+/// Only available to tests.
+EnvironmentOverride get xdgEnvironmentOverride => _xdgEnvironmentOverride;
+EnvironmentOverride _xdgEnvironmentOverride;
+EnvironmentOverride _productionGetEnv = (String value) => Platform.environment[value];
+EnvironmentOverride _getenv = _productionGetEnv;
+
+/// A testing function that replaces the process manager used to run xdg-user-path
+/// with the one supplied.
+///
+/// Only available to tests.
+@visibleForTesting
+set xdgProcessManager(ProcessManager processManager) {
+  _processManager = processManager;
+}
+
+ProcessManager _processManager = const LocalProcessManager();
+
+List<Directory> _directoryListFromEnvironment(String envVar, String fallback) {
```

Comment:
It seems like it would be clearer to make this `List<String> fallbacks`, and then just return it if value is null/empty without continuing to the splitting. There's no reason our fallback has to use a string representation of a list instead of an actual list.

Plus it would eliminate the confusion of this `fallback` and the next `fallback` being totally different things (one a list of absolute paths, the other a single relative path)